### PR TITLE
fix(DB/Quest/Creature): Poisoned Water: Transformed elementals.

### PR DESCRIPTION
**Pull request by febaut**
During the quest "Poisoned Water" in Eastern Plaguelands, you use the item [Aspect of Neptulon] to transform elemental in Discordant Surges. After the transformation the elementals didn't attack you. Now they do as it supposed.

## Changes Proposed:
-  Add a row to related creatures' SMART_SCRIPTS that pulls the creature on spellhit. 

## Issues Addressed:

- Closes https://github.com/azerothcore/azerothcore-wotlk/issues/9860

## Tests Performed:

- 
- 

## How to Test the Changes:

First apply the SQL scripts. Then restart the server (.reload smart might not apply the changes to spawned creatures).
1. .go c 92084
2. .quest add 6804
3. Use [Aspect of Neptulon] on elementals

## Known Issues and TODO List:

- [ ]
- [ ]

## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.

### CODE REVIEW

File: data/sql/updates/pending_db_world/rev_1641244560298659050.sql [hunk 1]
```diff
@@ -0,0 +1,4 @@
+INSERT INTO `version_db_world` (`sql_rev`) VALUES ('1641244560298659050');
+
+INSERT INTO `smart_scripts` (`entryorguid`, `source_type`, `id`, `link`, `event_type`, `event_phase_mask`, `event_chance`, `event_flags`, `event_param1`, `event_param2`, `event_param3`, `event_param4`, `event_param5`, `action_type`, `action_param1`, `action_param2`, `action_param3`, `action_param4`, `action_param5`, `action_param6`, `target_type`, `target_param1`, `target_param2`, `target_param3`, `target_param4`, `target_x`, `target_y`, `target_z`, `target_o`, `comment`) VALUES (8522, 0, 2, 0, 8, 0, 100, 0, 21332, 0, 0, 0, 0, 49, 0, 0, 0, 0, 0, 0, 7, 0, 0, 0, 0, 0, 0, 0, 0, 'Plague Monstrosity - On Spellhit \'Aspect of Neptulon\' - Start Attacking');
```
Comment (acidmanifesto) [inline data/sql/updates/pending_db_world/rev_1641244560298659050.sql:3]:
```
INSERT INTO `smart_scripts` (`entryorguid`, `source_type`, `id`, `link`, `event_type`, `event_phase_mask`, `event_chance`, `event_flags`, `event_param1`, `event_param2`, `event_param3`, `event_param4`, `event_param5`, `action_type`, `action_param1`, `action_param2`, `action_param3`, `action_param4`, `action_param5`, `action_param6`, `target_type`, `target_param1`, `target_param2`, `target_param3`, `target_param4`, `target_x`, `target_y`, `target_z`, `target_o`, `comment`) VALUES 
(8522, 0, 2, 0, 8, 0, 100, 0, 21332, 0, 0, 0, 0, 49, 0, 0, 0, 0, 0, 0, 7, 0, 0, 0, 0, 0, 0, 0, 0, 'Plague Monstrosity - On Spellhit \'Aspect of Neptulon\' - Start Attacking'),
(8521, 0, 3, 0, 8, 0, 100, 0, 21332, 0, 0, 0, 0, 49, 0, 0, 0, 0, 0, 0, 7, 0, 0, 0, 0, 0, 0, 0, 0, 'Blighted Horror - On Spellhit \'Aspect of Neptulon\' - Start Attacking');
```

Comment (acidmanifesto) [reply data/sql/updates/pending_db_world/rev_1641244560298659050.sql:3]: Condense two seperate queries into one query.
Line Break after VALUE

**Review by acidmanifesto — CHANGES_REQUESTED** [review]

**Comment by Nyeriah** [comment]
Hey, 

First of all, thank you very much for your contribution. However, this looks wrong to me as creatures shouldn't need an event call to start attacking players, this likely has to do with them having the wrong flags or wrong react states.

I'm not sure if we'll be able to get some sniffing on this, but that's most definitely the case for them.

Regards

**Comment by febaut** [comment]
> 
In this case the spell that does the creature template update has a maximum range of 30 yards which is more than threat range. If you are close enough to the mob it will attack, as a normal mob does.

**Comment by Kitzunu** [comment]
no

**Comment by Nyeriah** [comment]
Well, what I understood of this issue is that the transformed elementals never engaged players, but looking into that report the second image has a level 81 character so that would explain why it doesn't try to engage him.

>> In this case the spell that does the creature template update has a maximum range of 30 yards which is more than threat range. If you are close enough to the mob it will attack, as a normal mob does, then is not flags or something else.

This is likely wrong then, because if the creature should be put in combat with the player after being hit by it then the spell itself would handle that, but it seems this spell has no initial threat by design so that never happens.

**Comment by febaut** [comment]
Yep, I tried with lvl 56. I'm curious about the spell because it does basically nothing. The transformation is through Smart Script as well.

**Comment by acidmanifesto** [comment]
Added Need Sniff label for further clarification in how it should be handled.

**Pull request closed without merging**